root/static/scripts/release-editor/fields.js

```diff
@@ -267,24 +267,23 @@ class Track {
   }
 
   titleDiffersFromRecording() {
-    return this.name() !== this.recording().name;
+    return this.hasExistingRecording() &&
+      this.name() !== this.recording().name;
   }
 
   artistDiffersFromRecording() {
-    const recording = this.recording();
-
     /*
      * This function is used to determine whether we can update the
      * recording AC, so if there's no recording, then there's nothing
      * to compare against.
      */
-    if (!recording || !recording.gid) {
```

Comment:
As far as I can see, `this.recording()` never returns null, so it seems simpler to just call `hasExistingRecording` here.